firebase-firestore/src/main/java/com/google/firebase/firestore/local/QueryContext.java

```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+/** A tracker to keep a record of important details during database local query execution. */
+public class QueryContext {
```

Comment:
Will `QueryContext` objects be passed between threads anywhere? If yes, make `documentReadCount` an `AtomicInteger` instead of a plain-old `int`.

Reply:
Cheryl, can you confirm whether `QueryContext` instances will be shared between threads? I think the answer is "no", but I'd like you to confirm, because if it _is_ shared between threads then the `int` member variables needs some protection to ensure proper synchronization between threads.

Reply:
`documentReadCount` will not be shared among multi threads. The only place calling it is in the main thread. @wu-hui please correct me if I am wrong.
```
db.query(sql.toString())
        .binding(bindVars)
        .forEach(
            row -> {
              processRowInBackground(backgroundQueue, results, row, filter);
              if (context != null) {
                context.incrementDocumentReadCount();
              }
            });
            ```

Reply:
As long as the lambda specified to `forEach` is called by the same thread as `db.query` (which I imagine it is).

Reply:
Yeah, they are from the same thread.

Reply:
Ok thanks for confirming. I'm now satisifed that `QueryContext` can safely use `int` rather than `AtomicInteger`.